backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,211 @@
+     * @param settings
+     */
+    public BacktraceEventsHandler(Context context, Map<String, Object> customAttributes,
+                                  Api api, final BacktraceHandlerThread backtraceHandlerThread, String urlPrefix, BacktraceMetricsSettings settings) {
+        // This should always have a nonnull looper because BacktraceHandlerThread starts in the
+        // constructor and getLooper blocks until the looper is ready if the thread is started
+        //
+        // We cannot cache the looper here since the super call has to go first (i.e: before
+        // declaring and assigning a Looper variable)
+        super(backtraceHandlerThread.getLooper());
+        if (backtraceHandlerThread.getLooper() == null) {
+            throw new NullPointerException("Expected nonnull looper, this should not happen");
+        }
+        this.context = context;
+        this.customAttributes = customAttributes;
+        this.backtraceHandlerThread = backtraceHandlerThread;
+        this.api = api;
+        this.submissionUrl = settings.getBaseUrl() + "/" + urlPrefix + "/submit?token=" + settings.getToken() + "&universe=" + settings.getUniverseName();
+        this.timeBetweenRetriesMillis = settings.getTimeBetweenRetriesMillis();
+
+        long timeIntervalMillis = settings.getTimeIntervalMillis();
+
+        BacktraceAttributes backtraceAttributes = new BacktraceAttributes(context, null, null);
+        this.application = backtraceAttributes.getApplicationName();
+        this.appVersion = backtraceAttributes.getApplicationVersionOrEmpty();
+
+        if (timeIntervalMillis != 0) {
+            final BacktraceEventsHandler handler = this;
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    handler.send();
+                    handler.postDelayed(this, timeIntervalMillis);
+                }
+            }, timeIntervalMillis);
+        }
+    }
+
+    /**
+     * Number of events in the queue
+     */
+    public int getCount() {
+        return events.size();
+    }
+
+    public int getMaximumNumberOfEvents() {
+        return this.maximumNumberOfEvents;
+    }
+
+    public void setMaximumNumberOfEvents(int maximumNumberOfEvents) {
+        this.maximumNumberOfEvents = maximumNumberOfEvents;
+    }
+
+    public abstract void sendStartupEvent(String eventName);
+
+    public void send() {
+        if (events == null) {
+            return;
+        }
+        if (events.size() == 0) {
+            return;
+        }
+        sendEvents(events);
+    }
+
+    protected abstract void sendEvents(ConcurrentLinkedDeque<T> events);
+
+    protected abstract void sendEventsPayload(EventsPayload<T> payload);
+
+    @Override
+    public void handleMessage(Message msg) {
+        final BacktraceHandlerInputEvents input = (BacktraceHandlerInputEvents) msg.obj;
+        EventsResult result;
+
+        if (input.eventsRequestHandler != null) {
```

Comment:
move this if-else statement to private method `getEventResult()`

Reply:
Will do!